client/src/app/components/common/save-as-modal.component.ts

```diff
@@ -43,15 +43,25 @@ import {ModalService} from "../modal";
 `
 })
 export class SaveAsModalComponent {
+
+    /** Path of the source file that should be copied */
+    public filePath: string;
+
+    /** Toggles the cog loader */
     private isCreatingFile: boolean;
+
+    /** Holds the error object that should be shown below the form if exists */
     private error: {[message: string]: string};
+
+    /** New file destination entry form */
     private newFileForm: ControlGroup;
 
     constructor(private formBuilder: FormBuilder,
                 private eventHub: EventHubService,
-                private workspace: WorkspaceService,
                 private modal: ModalService) {
 
+        this.filePath = undefined;
```

Comment:
Great question.
Typescript doesn't emit JS properties until they are used, which makes a difference between a property that *IS* undefined, and the one that actually doesn't exist even as an undefined. Philosophical equivalent to something that just *ISN'T*. 
Only when `this.filePath` is used for the first time, it becomes an **enumerable** property with *undefined* value.  
This particular component is dynamically created for the modal, and `filePath` is a public property that is to be set from an external source at the time of creation. For those cases ([example](https://github.com/ivanbatic/cottontail/blob/dfe47b72235af4643e8eb8675843568e659f6c96/client/src/app/components/modal/modal.component.ts#L134)), I'm using the `Chap.applyParams` ([def](https://github.com/ivanbatic/cottontail/blob/dfe47b72235af4643e8eb8675843568e659f6c96/client/src/app/helpers/chap.ts#L11)) function that copies over all values from a given source object to equivalent **enumerable** properties of a destination object.  
So, if a prop is not there, it will not be copied over. The reason for this is that I don't want to end up with an object that has any property or method that is not explicitly declared wherever that object is created (or class defined).

If you or @mayacoda have a more elegant solution, I'll gladly refactor this.